src/components/Firestore/DocumentEditor/NumberEditor.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { TextField } from '@rmwc/textfield';
+
+const NumberEditor: React.FC<{
+  value: number;
+  onChange: (value: number) => void;
+}> = ({ value, onChange }) => {
+  const [state, setState] = useState<string | number>(value);
+
+  useEffect(() => {
+    setState(value);
+  }, [value, setState]);
+
+  function handleChange(value: string) {
+    const newValue = parseInt(value);
```

Comment:
`parseInt` truncates decimals.

```js
Number('1234.5');
```

Reply:
Good catch, switched to parseFloat... I'm not using Number since it returns 0 for non-parsed values.

Reply:
`parseFloat("1.2garbage")` will happily return `1.2` though. `Number` on the other hand gives you `NaN` for complete garbage or trailing garbage.